src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -453,20 +520,19 @@ private void AddTasksForExecutingSymbolActions(SymbolDeclaredCompilationEvent sy
             }
         }
 
-        protected static void AddAnalyzerActionsExecutor(IDictionary<DiagnosticAnalyzer, Task> map, DiagnosticAnalyzer analyzer, Action executeAnalyzerActions, CancellationToken cancellationToken)
+        protected static void AddAnalyzerActionsExecutor(IDictionary<DiagnosticAnalyzer, ArrayBuilder<Action>> map, DiagnosticAnalyzer analyzer, Action executeAnalyzerActions)
         {
```

Comment:
Yes, this is just computing what actions need to execute, not actual execution. Its done on a single thread as it should be quick.